includes/class-indieauth-client-discovery.php

```diff
@@ -74,45 +75,63 @@ private function parse( $url ) {
 			return $response;
 		}
 
-		$content = wp_remote_retrieve_body( $response );
-
-		if ( class_exists( 'Masterminds\\HTML5' ) ) {
-			$domdocument = new \Masterminds\HTML5( array( 'disable_html_ns' => true ) );
-			$domdocument = $domdocument->loadHTML( $content );
-		} else {
-			$domdocument = new DOMDocument();
-			libxml_use_internal_errors( true );
-			if ( function_exists( 'mb_convert_encoding' ) ) {
-				$content = mb_convert_encoding( $content, 'HTML-ENTITIES', mb_detect_encoding( $content ) );
+		$content_type = wp_remote_retrieve_header( $response, 'content-type' );
```

Comment:
`parse()` seems like it should be designated as `static` since that's how it is used. Also it seems as though it could get to the bottom and not `return` so I think it needs a return at the bottom. I also suggest a doc block describe the params and what it does if there's an error.